src/sparseml/pytorch/utils/quantization/quantize_qat_export.py

```diff
@@ -508,21 +516,219 @@ def _convert_quantizable_gemm(
         remove_node_and_params_from_graph(model, gemm_node, keep_params=params_to_keep)
 
 
+def _convert_quantizable_matmul_and_add(model: ModelProto):
+    """
+    A pass for converting a MatMul with kernel and bias into a quantized representation
+
+    Starting with:
+             INPUT         QuantizeLinear (with constant kernel)
+               |               |
+        QuantizeLinear     DequantizeLinear
+               |               |
+        DequantizeLinear   Transpose
+                     |      |
+                      MatMul
+                        |
+                       Add (with constant bias)
+                        |
+                  QuantizeLinear
+                        |
+                 DequantizeLinear
+                        |
+                     OUTPUT
+
+    We end up converting to:
+          INPUT
+            |
+        QuantizeLinear
+            |
+        QLinearMatMul (with constant kernel)
+            |
+        QLinearAdd (with constant bias)
+            |
+        DequantizeLinear
+            |
+          OUTPUT
+    """
+    matmul_nodes = [n for n in model.graph.node if n.op_type in ["MatMul"]]
```

Comment:
could we add some debug logging to this function? seems very long-running and potentially hard to debug

Reply:
yeah added an info count and debug for each matmul
```
2021-04-19 15:51:55 sparseml.pytorch.utils.quantization.quantize_qat_export INFO     Converted 73 quantizable MatMul ops with weight and bias to QLinearMatMul and QLinearAdd
```